docs/guides/extension_points.rst

```diff
@@ -0,0 +1,165 @@
+|                     |                                          | contexts, to affect standard pages delivered by the core    |
+|                     |                                          | platform. See `Plugin Contexts`_ to learn more.             |
++---------------------+------------------------------------------+-------------------------------------------------------------+
+| Course tab          | ``openedx.course_tab``                   | A course tab plugin adds a new tab shown to learners within |
+|                     |                                          | a course. ``courseware``, ``course_info``, and              |
+|                     |                                          | ``discussion`` are examples of built-in tab plugins.        |
+|                     |                                          | Read the `course tabs documentation`_ to learn more.        |
++---------------------+------------------------------------------+-------------------------------------------------------------+
+| Course tool         | ``openedx.course_tool``                  | The course home page (the landing page for the course)      |
+|                     |                                          | includes a "Course Tools" section that provides links to    |
+|                     |                                          | "tools" associated with the course. Examples of course tool |
+|                     |                                          | plugins included in the core are reviews, updates, and      |
+|                     |                                          | bookmarks. See |course_tools.py|_ to learn more.            |
++---------------------+------------------------------------------+-------------------------------------------------------------+
+| Custom registration | ``REGISTRATION_EXTENSION_FORM`` Django   | By default, the registration page for each instance of Open |
+| form app            | setting (LMS).                           | edX has fields that ask for information such as a user’s    |
+|                     |                                          | name, country, and highest level of education completed.    |
+|                     |                                          | You can add custom fields to the registration page for your |
+|                     |                                          | own Open edX instance. These fields can be different types, |
+|                     |                                          | including text entry fields and drop-down lists. See        |
+|                     |                                          | `Adding Custom Fields to the Registration Page`_.           |
++---------------------+------------------------------------------+-------------------------------------------------------------+
+| Learning Context    | ``openedx.learning_context``             | A "Learning Context" is a course, a library, a program, a   |
```

Comment:
> This is great. I hadn't realized that this was built in a generic way for others to use.
Q: Could this be further generalized and extracted out of edx-platform so it can be used in other repos/services? I believe the upcoming LTI Advantage development effort could make use of this.

Thanks :)

Yes, any service can be designed to think in terms of learning contexts instead of hard-coding to use "courses", and the `opaque-keys` repo includes [`LearningContextKey`](https://github.com/edx/opaque-keys/blob/f90e5fdc67f4490d894dd286a824da51a2821e21/opaque_keys/edx/keys.py#L13-L39) and defines `CourseKey` as a subclass of it. I agree it's a good fit for LTI, and I believe the LTI spec uses "context" rather than "course" anyways.